guest/rust/api_core/src/client/material.rs

```diff
@@ -0,0 +1,34 @@
+use crate::global::{ProceduralMaterialHandle, ProceduralSamplerHandle, ProceduralTextureHandle};
+use crate::internal::conversion::*;
+use crate::internal::wit;
+
+#[derive(Clone, Copy)]
+pub struct Descriptor {
+    pub base_color_map: ProceduralTextureHandle,
```

Comment:
I wonder... should these be procedural texture handles, or just texture handles? What if I want to use a texture in my `assets/` with a procedural material?

Reply:
Maybe out of scope for this initial version, but I do wonder how we'd handle this

Reply:
Good question. We briefly discussed this idea where instead of `String` urls, the asset pipeline code generates IDs for each asset, something like `pub const MY_COLOR_TEXTURE: SomeIDType = ...;`. At load time, the engine resolves this ID into the physical location of the asset, which could be in a local directory or at a remote server. This indirection would make asset URLs slightly more efficient since it's no longer an arbitrarily long string, and it would also reduce the chance of human error in typing out string URLs, and also it might be more resilient to directory changes. One downside though is the fact that it is an indirection, so it might be harder to understand.

Reply:
Once we work on custom shaders I think I'll change it to `TextureHandle` to match `BufferHandle`.